hudi-sync/hudi-hive-sync/pom.xml

```diff
@@ -200,6 +200,9 @@
             </goals>
           </execution>
         </executions>
+        <configuration>
+          <skip>false</skip>
+        </configuration>
```

Comment:
true or false ?

Reply:
false.https://maven.apache.org/plugins/maven-jar-plugin/test-jar-mojo.html#skip

Reply:
> true or false ?
false
https://maven.apache.org/plugins/maven-jar-plugin/test-jar-mojo.html#skip

Reply:
So the purpose is to generate test jars for these two modules.

Reply:
Yes. Because these two module test classes are referenced elsewhere, such as in the hudi cli module, which introduces hudi utilities test jar

<dependency>
<groupId>org. apache. hudi</groupId>
<artifactId>hudi utilities_$ {scala. binary. version}</artictId>
<version>${project. version}</version>
<scope>test</scope>
<type>test jar</type>
</dependency>

Reply:
Did you introduce dependencies for these test jars in your own project?

Reply:
No. The problem occurred after I cloned the project.

Reply:
Of course, you can also check the project to see if the dependency using the test jar has added the skip tag.

Reply:
I'm still confused what the patch want to fix, is it a compilure failure or test failure or something else?

Reply:
hello.

Reply:
Hello，when I compiled the hudi project, there was a dependency issue where I couldn't find it. `[ERROR] Failed to execute goal on project hudi-utilities_2.12: Could not resolve dependencies for project org.apache.hudi:hudi-utilities_2.12:jar:0.14.0-SNAPSHOT: Could not find artifact org.apache.hudi:hudi-hive-sync:jar:tests:0.14.0-SNAPSHOT in confluent (https://packages.confluent.io/maven/) -> [Help 1]`

Reply:
Hello，when I compiled the hudi project, there was a dependency issue where I couldn't find it.

mvn clean package -Dmaven.test.skip=true -Dcheckstyle.skip=true -Drat.skip=true 

[ERROR] Failed to execute goal on project hudi-utilities_2.12: Could not resolve dependencies for project org.apache.hudi:hudi-utilities_2.12:jar:0.14.0-SNAPSHOT: Could not find artifact org.apache.hudi:hudi-hive-sync:jar:tests:0.14.0-SNAPSHOT in confluent (https://packages.confluent.io/maven/) -> [Help 1]

Reply:
Did you compile the project under the root dir?

Reply:
yes.

Reply:
Yes. You can take a look at the abnormal descriptions in the above chat content.

Reply:
Weird, I can not reproduce it, maybe it is because of your local mvn repository env.

Reply:
I'm also surprised that theoretically, this problem is inevitable. When compiling, the hudi cli module references the hudi utilities test jar, which is configured through the skip tag.My Compilation Command: mvn clean package -Dmaven.test.skip=true -Dcheckstyle.skip=true -Drat.skip=true

Reply:
> Weird, I can not reproduce it, maybe it is because of your local mvn repository env.

hello, I also encountered this problem. when I use this command to compile project.  
`mvn clean install -DskipTests -Dscala-2.12 -Dspark3.2 -Dmaven.test.skip=true -Dcheckstyle.skip=true -Dflink1.16 -Drat.skip=true`

`[ERROR] Failed to execute goal on project hudi-utilities_2.12: Could not resolve dependencies for project org.apache.hudi:hudi-utilities_2.12:jar:0.15.0-SNAPSHOT: org.apache.hudi:hudi-hive-sync:jar:tests:0.15.0-SNAPSHOT was not found in https://packages.confluent.io/maven/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of confluent has elapsed or updates are forced -> [Help 1]
`

This test jar, I don't know how to generate it.

Reply:
And when I add this two configs, it was fixed.

Reply:
<plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
        <version>3.1.2</version>
        <executions>
          <execution>
            <goals>
              <goal>test-jar</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <skip>false</skip>
        </configuration>
      </plugin>

Reply:
`<plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
        <version>3.1.2</version>
        <executions>
          <execution>
            <goals>
              <goal>test-jar</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <skip>false</skip>
        </configuration>
      </plugin>` generate test jar.

Reply:
Thanks, can we make the tests pass.